ckanext/canada/view.py

```diff
@@ -202,6 +202,15 @@ def post(self, package_type, id):
 
 
 class CanadaUserRegisterView(UserRegisterView):
+    def get(self, data=None, errors=None, error_summary=None):
+        # additional check to ensure user can access the Request an Account page
+        # only possible if accessing from GOC network
+        remote_addr = request.headers.get('X-Forwarded-For') or \
+                      request.environ.get('REMOTE_ADDR')
+        if not h.registry_network_access(remote_addr):
+            abort(403, _('Not authorized to see this page'))
+        return super(CanadaUserRegisterView, self).get()
+
```

Comment:
This check should be done in an overridden user_create auth function. That way it applies to the views, links and APIs equally